### PR TITLE
bug/59120 - upgrade to typescript 4.5 & eslint 34

### PR DESCRIPTION
**Pull request by GuyHarwood**
introduces more typescript rules and aligns better with codacy

### CODE REVIEW

File: pupil-api/yarn.lock [hunk 1]
```diff
@@ -17,7 +17,7 @@
   dependencies:
     tslib "^2.2.0"
 
-"@azure/core-amqp@^3.3.0":
+"@azure/core-amqp@^3.1.0":
```
Comment (jon-shipley) [inline pupil-api/yarn.lock:20]: Why the downgrade?

**Review by jon-shipley — APPROVED** [review]
LGTM :)

Many thanks for this improvement!  Just one comment on a package downgrade.

File: pupil-api/yarn.lock (hunk 1 above)
Comment (GuyHarwood) [reply pupil-api/yarn.lock:20]: 👀  well that's odd, given i didn't touch that app.

I need to align all of the projects onto TS 4.5.5, so will focus on that next.

**Comment by GuyHarwood** [comment]
@jon-shipley merge conflicts resolved, please re-review

**Comment by GuyHarwood** [comment]
Parking until 2023 check window closes

**Pull request merged**